server/src/main/java/org/elasticsearch/gateway/MetaDataStateFormat.java

```diff
@@ -177,20 +176,35 @@ private static void performStateDirectoriesFsync(List<Tuple<Path, Directory>> st
         }
     }
 
+    /**
+     * Writes the given state to the given directories and performs cleanup.
+     * See also {@link #write(Object, Path...)} and {@link #cleanupOldFiles(long, Path[])}.
+     */
+    public final long writeAndCleanup(final T state, final Path... locations) throws WriteStateException {
```

Comment:
Well, previously I've had just write method that accepts true/false, but David asked me to add a new method writeAndCleanup. Could be you please think once again about your preference, so that I don't go back and forward on this and I'll implement that solution?

Reply:
Happy to hear @DaveCTurner's thoughts on this. Note that my suggestion is different to the boolean true / false, and I think it merits its own consideration.

Reply:
I also think this should be two methods. If one of them were to be called `write` then I would slightly prefer that one not to do the cleanup, but I do not hold that opinion strongly.

Reply:
I agree with David that if we don't want to change the old name - "write", it would be confusing that write method performs cleanup, even if we have "writeAndNoCleanup" method. So my proposal is to leave as it is.

Reply:
ok